include/MySQL_Thread.h

```diff
@@ -365,19 +356,8 @@ class MySQL_Threads_Handler
 		char *init_connect;
 		char *ldap_user_variable;
 		char *add_ldap_user_comment;
-		char *default_sql_mode;
-		char *default_time_zone;
-		char *default_isolation_level;
-		char *default_transaction_read;
 		char *default_tx_isolation;
-		char *default_character_set_results;
-		char *default_session_track_gtids;
-		char *default_sql_auto_is_null;
-		char *default_sql_select_limit;
-		char *default_sql_safe_updates;
-		char *default_collation_connection;
-		char *default_net_write_timeout;
-		char *default_max_join_size;
+		char *default_variables[SQL_NAME_LAST];
```

Comment:
I checked how this will work with `std::array` and actually it works. I just added 
```
std::array<char*,SQL_NAME_LAST> deafult_variables;
```
The code was compiled and tests have been passed.

However, I do not see much benefits of converting the code. The two advantages that it has:
1. `std::array` has `size()` member
2. we can use C++ algorithms

are not critical here, because we do not use algorithms with `default_variables` and we always have global definition `SQL_NAME_LAST` to use instead of `size()`

The change to `std::array` is more stylistic change and so should be applied to all C-style arrays to keep code style consistent through all files. Otherwise code become less understandable. Technically I do not see much benefits here.